src/python/bot/fuzzers/syzkaller/fuzzer.py

```diff
@@ -0,0 +1,125 @@
+VMLINUX_FOLDER = '/tmp/syzkaller/vmlinux'
+
+
+def get_arguments(unused_fuzzer_path):
+  """Get arguments for a given fuzz target."""
+  build_dir = environment.get_value('BUILD_DIR')
+  device_serial = environment.get_value('ANDROID_SERIAL')
+  json_config_path = os.path.join('/tmp', device_serial, 'config.json')
+  config.generate(
+      serial=device_serial,
+      work_dir_path=SYZKALLER_WORK_FOLDER,
+      binary_path=os.path.join(build_dir, 'syzkaller'),
+      vmlinux_path=VMLINUX_FOLDER,
+      config_path=json_config_path,
+      kcov=True,
+      reproduce=False)
+  return ['--config', json_config_path]
+
+
+def get_runner(fuzzer_path):
+  """Return a suzkaller runner object."""
+  build_dir = environment.get_value('BUILD_DIR')
+  return AndroidSyzkallerRunner(fuzzer_path, build_dir)
+
+
+class AndroidSyzkallerRunner(new_process.ProcessRunner):
+  """Syzkaller runner."""
+
+  def __init__(self, executable_path, default_args=None):
+    """Inits the AndroidSyzkallerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    super(AndroidSyzkallerRunner, self).__init__(
+        executable_path=executable_path, default_args=None)
+
+  def get_command(self, additional_args=None):
+    """Process.get_command override."""
+    base_command = super(AndroidSyzkallerRunner,
+                         self).get_command(additional_args=additional_args)
+
+    return base_command
+
+  def _create_empty_testcase_file(self):
+    """Create an empty testcase file in temporary directory."""
+    _, path = tempfile.mkstemp(dir=fuzzer_utils.get_temp_dir())
+    return path
+
+  def get_testcase_path(self, log_lines):
+    """Get testcase path from log lines."""
+    #TODO(hzawawy) when a crash is detected extract testcase from report.
+    for line in log_lines:
+      match = re.match(KASAN_CRASH_TESTCASE_REGEX, line)
+      if match:
+        return match.group(1)
+
+    return None
+
+  def fuzz(self,
+           fuzz_timeout,
+           additional_args,
+           unused_additional_args=None,
+           unused_extra_env=None):
+    """This is where actual syzkaller fuzzing is done."""
+    additional_args = copy.copy(additional_args)
+    fuzz_result = self.run_and_wait(additional_args, timeout=fuzz_timeout)
+
+    log_lines = utils.decode_to_unicode(fuzz_result.output).splitlines()
+    fuzz_result.output = None
+    crash_testcase_file_path = self.get_testcase_path(log_lines)
+
+    #TODO(hzawawy): remove once syzkaller code is completed.
+    if not crash_testcase_file_path and fuzz_result.return_code:
+      crash_testcase_file_path = self._create_empty_testcase_file()
+
+    fuzz_logs = '\n'.join(log_lines)
+
+    # TODO(hzawawy): Parse stats information and add them to FuzzResult
+    parsed_stats = []
+
+    crashes = []
+    if crash_testcase_file_path:
+      #TODO(hzawawy): add repro arguments
+      reproduce_arguments = None
```

Comment:
This will probably cause an exception. Set it to `[]` instead.

Reply:
regarding first comment, Syzkaller is both the engine and the fuzzer, so this section does not apply to syzkaller.
regarding the second comment, done.